roles/custom/matrix-nginx-proxy/templates/nginx/conf.d/matrix-domain.conf.j2

```diff
@@ -94,6 +94,96 @@
 	}
 	{% endif %}
 
+	{% if matrix_nginx_proxy_proxy_media_repo_enabled %}
+	# Redirect all media endpoints to the media-repo
+	location ^~ /_matrix/media {
+		{% if matrix_nginx_proxy_enabled %}
+			{# Use the embedded DNS resolver in Docker containers to discover the service #}
+			resolver {{ matrix_nginx_proxy_http_level_resolver }} valid=5s;
+			set $backend "{{ matrix_nginx_proxy_proxy_media_repo_addr_with_container }}";
+			proxy_pass http://$backend;
+		{% else %}
+			{# Generic configuration for use outside of our container setup #}
+			proxy_pass http://{{ matrix_nginx_proxy_proxy_media_repo_addr_sans_container }};
+		{% endif %}
+
+		# Make sure this matches your homeserver in media-repo.yaml
+		# You may have to manually specify it if using delegation or the
+		# incoming Host doesn't match.
+		proxy_set_header Host $host;
+
+		proxy_set_header X-Real-IP $remote_addr;
+		proxy_set_header X-Forwarded-For $remote_addr;
+	}
+
+	# Redirect other endpoints registered by the media-repo to its container
+	# /_matrix/client/r0/logout
+	# /_matrix/client/r0/logout/all
+	location ^~ /_matrix/client/(r0|v1|v3|unstable)/(logout|logout/all) {
+		{% if matrix_nginx_proxy_enabled %}
+			{# Use the embedded DNS resolver in Docker containers to discover the service #}
+			resolver {{ matrix_nginx_proxy_http_level_resolver }} valid=5s;
+			set $backend "{{ matrix_nginx_proxy_proxy_media_repo_addr_with_container }}";
+			proxy_pass http://$backend;
+		{% else %}
+			{# Generic configuration for use outside of our container setup #}
+			proxy_pass http://{{ matrix_nginx_proxy_proxy_media_repo_addr_sans_container }};
+		{% endif %}
+
+		# Make sure this matches your homeserver in media-repo.yaml
+		# You may have to manually specify it if using delegation or the
+		# incoming Host doesn't match.
+		proxy_set_header Host $host;
+
+		proxy_set_header X-Real-IP $remote_addr;
+		proxy_set_header X-Forwarded-For $remote_addr;
+	}
```

Comment:
These seem like fairly generic API endpoints to capture and forward to matrix-media-repo.

Where is this coming from? I don't see it one their nginx example: https://docs.t2bot.io/matrix-media-repo/deployment/nginx.html

Reply:
Yeah, the proxy configuration documentation unfortunately does not mention configuring those endpoints. The full list of endpoints used by the application can be found here: https://github.com/turt2live/matrix-media-repo/blob/release-v1.2.8/api/webserver/webserver.go#L85

The reason for proxying the endpoints through the media repo however is for caching access tokens. Details regarding that can be found here: https://docs.t2bot.io/matrix-media-repo/access-token-cache/index.html

Reply:
The relevant quote about the access token cache:

> The access token cache is disabled by default, however if enabled it is strongly recommended to proxy the /logout and /logout/all endpoints through to the media repo. They'll be proxied through to the homeserver and update the cache internally.

Reply:
Makes sense!

The only trouble I see is that matrix-corporal (if enabled), will no longer be able to see these calls. They'll go to media-repo and then straight to the homeserver. Unless... we configure matrix-media-repo to forward requests to matrix-corporal (adjusting `csApi` in `matrix_media_repo_homeservers`).

Speaking of which.. for `csApi` in `matrix_media_repo_homeservers`, shouldn't we (couldn't we) use some container URL (`matrix_homeserver_container_url` or maybe `matrix_nginx_proxy_proxy_matrix_client_api_addr_with_container` which is matrix-corporal-aware) instead of a public endpoint? I'm not sure if that would work or if it introduces some other problems. I wonder if matrix-media-repo uses the `csApi` for some public stuff or just internally.

Reply:
I did not test with matrix-corporal enabled, but replacing `csApi` with `matrix_nginx_proxy_proxy_matrix_client_api_addr_with_container` seems to work without issue with it disabled, expanding to the proxy URL  `http://matrix-nginx-proxy:12080`